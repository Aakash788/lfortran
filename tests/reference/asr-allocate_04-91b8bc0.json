{
    "basename": "asr-allocate_04-91b8bc0",
    "cmd": "lfortran --indent --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/allocate_04.f90",
    "infile_hash": "c1f0a934c2f96f63899ac8089408d00e764052695666f27ae6ced6c4",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-allocate_04-91b8bc0.stdout",
    "stdout_hash": "530a9ce11e33c86c24ff22faab7075b5843d7d766043452d912a1aea",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}
