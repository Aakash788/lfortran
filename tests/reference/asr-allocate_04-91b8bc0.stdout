(TranslationUnit
    (SymbolTable
        1
        {
            allocate_04:
                (Program
                    (SymbolTable
                        2
                        {
                            c:
                                (Variable
                                    2
                                    c
                                    []
                                    Local
                                    ()
                                    ()
                                    Allocatable
                                    (Integer 4 [(()
                                    ())
                                    (()
                                    ())
                                    (()
                                    ())])
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            len_trim:
                                (ExternalSymbol
                                    2
                                    len_trim
                                    4 len_trim
                                    lfortran_intrinsic_string
                                    []
                                    len_trim
                                    Private
                                ),
                            message:
                                (Variable
                                    2
                                    message
                                    []
                                    Local
                                    ()
                                    ()
                                    Allocatable
                                    (Character 1 -2 () [])
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            num:
                                (Variable
                                    2
                                    num
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Character 1 20 () [])
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            string:
                                (Variable
                                    2
                                    string
                                    []
                                    Local
                                    ()
                                    ()
                                    Allocatable
                                    (Character 1 -2 () [])
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    allocate_04
                    []
                    [(Allocate
                        [((Var 2 string)
                        [])]
                        ()
                        ()
                        ()
                    )
                    (Allocate
                        [((Var 2 message)
                        [])]
                        ()
                        ()
                        ()
                    )
                    (Allocate
                        [((Var 2 c)
                        [((IntegerConstant 1 (Integer 4 []))
                        (IntegerConstant 3 (Integer 4 [])))
                        ((IntegerConstant 1 (Integer 4 []))
                        (IntegerConstant 3 (Integer 4 [])))
                        ((IntegerConstant 1 (Integer 4 []))
                        (IntegerConstant 3 (Integer 4 [])))])]
                        ()
                        ()
                        ()
                    )
                    (=
                        (Var 2 string)
                        (StringConstant
                            "lfortran"
                            (Character 1 8 () [])
                        )
                        ()
                    )
                    (If
                        (StringCompare
                            (Var 2 string)
                            NotEq
                            (StringConstant
                                "lfortran"
                                (Character 1 8 () [])
                            )
                            (Logical 4 [])
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (=
                        (Var 2 num)
                        (StringConstant
                            "lfortran working"
                            (Character 1 16 () [])
                        )
                        ()
                    )
                    (=
                        (Var 2 message)
                        (StringSection
                            (Var 2 num)
                            (IntegerConstant 1 (Integer 4 []))
                            (IntegerBinOp
                                (FunctionCall
                                    2 len_trim
                                    ()
                                    [((Var 2 num))]
                                    (Integer 4 [])
                                    ()
                                    ()
                                )
                                Add
                                (IntegerConstant 1 (Integer 4 []))
                                (Integer 4 [])
                                ()
                            )
                            (IntegerConstant 1 (Integer 4 []))
                            (Character 1 -1 () [])
                            ()
                        )
                        ()
                    )
                    (Print
                        ()
                        [(Var 2 message)]
                        ()
                        ()
                    )
                    (ImplicitDeallocate
                        [2 c
                        2 message
                        2 string]
                    )]
                ),
            iso_fortran_env:
                (IntrinsicModule lfortran_intrinsic_iso_fortran_env),
            lfortran_intrinsic_builtin:
                (IntrinsicModule lfortran_intrinsic_builtin),
            lfortran_intrinsic_string:
                (IntrinsicModule lfortran_intrinsic_string)
        })
    []
)
